site.yml

```diff
@@ -78,6 +78,36 @@
     - role: percona-backup
       tags: ['infra', 'percona']
 
+- name: install mongodb on primary
+  gather_facts: force
+  hosts: mongo_db[0]
+  vars:
+    mongodb_primary: True
+  roles:
+    - role: mongodb
+      tags: ['mongodb', 'mongodb-primary']
+      when: ceilometer.enabled|bool
+
+- name: install mongodb on secondaries
+  gather_facts: force
+  hosts: mongo_db:!mongo_db[0]
+  vars:
+    mongodb_primary: False
+  roles:
+    - role: mongodb
+      tags: ['mongodb', 'mongodb-secondary']
+      when: ceilometer.enabled|bool
```

Comment:
Each task is ran in parallel, however there are controls to make sure the task only runs on one host. Can you pin point the tasks that would fail if each task was ran in parallel?